interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -167,6 +167,8 @@ public abstract class AbstractInteropTest {
       new AtomicReference<>();
   private final AtomicReference<ClientCall<?, ?>> clientCallCapture =
       new AtomicReference<>();
+  private static final ThreadLocal<ClientCall<?, ?>> threadClientCallCapture =
```

Comment:
nit: instead of using a `ThreadLocal` can we just declare a client call capture local to each thread? I.e. local to `executeSoakTestInThread` ?

I think we can get rid of `recordThreadClientCallInterceptor` and just use `recordClientCallInterceptor`, then